sdk/core/core/inc/az_span.h

```diff
@@ -110,77 +92,54 @@ AZ_NODISCARD AZ_INLINE int32_t az_span_capacity(az_span span) { return span._int
  */
 #define AZ_SPAN_FROM_STR(STRING_LITERAL) (az_span) AZ_SPAN_LITERAL_FROM_STR(STRING_LITERAL)
 
-/**
- * @brief az_span_init returns a span over a byte buffer.
- *
- * @param[in] ptr The memory address of the 1st byte in the byte buffer
- * @param[in] length The number of bytes initialized in the byte buffer
- * @param[in] capacity The number of total bytes in the byte buffer
- * @return az_span The "view" over the byte buffer.
- */
-AZ_NODISCARD az_span az_span_init(uint8_t* ptr, int32_t length, int32_t capacity);
-
-/**
- * @brief az_span_from_str returns an az_span from a 0-terminated array of bytes (chars)
- *
- * @param[in] str The pointer to the 0-terminated array of bytes (chars)
- * @return az_span An az_span over the byte buffer; length & capacity are set to the string's
- * length not including the \0 terminator.
- */
-AZ_NODISCARD az_span az_span_from_str(char* str);
+// Returns 1 if the address of the array is equal to the address of its 1st element
+// https://stackoverflow.com/questions/16794900/validate-an-argument-is-array-type-in-c-c-pre-processing-macro-on-compile-time
```

Comment:
Remove this http link please

Reply:
I read the link, and I don't see where you go this formula from.

Reply:
I only see this return a 0, else a compiler error

Reply:
Is there a unit test for this?

Reply:
I added a unit test and comments to help clarify things. We can't add unit tests to directly see what the `AZ_SPAN_FROM_BUFFER` macro does because most new compilers would just fail at compile time, such as msvc VS 2019, clang 10.0.0, gcc 9.3, etc. (while some older ones would fail at runtime or have undefined behavior for division by 0).